zug/transducer/filter.hpp

```diff
@@ -29,11 +71,15 @@ namespace zug {
 template <typename PredicateT>
 auto filter(PredicateT&& predicate)
 {
-    return comp([=](auto&& step) {
-        return [=, p = predicate](auto&& s, auto&&... is) mutable {
-            return compat::invoke(p, is...)
-                       ? call(step, ZUG_FWD(s), ZUG_FWD(is)...)
-                       : skip(step, ZUG_FWD(s), ZUG_FWD(is)...);
+    return comp([p = ZUG_FWD(predicate)](auto&& step) {
+        return [p = std::move(p), step = ZUG_FWD(step)](auto&& s,
```

Comment:
Is it correct that these 2 `return`s are never called multiple times and only third `return`, the actual filtering code, is called on each iteration?. If not, I need to remove forwarding.

Reply:
What about this? I might open a separate PR. It caught my eye because some arguments passed by forward reference aren't forwarded anywhere.